url8db16c7d62_Arsaell_EdT_tgz-pJ8-DATA_MainJ8/src/GUI/MainFrame.java

```diff
@@ -0,0 +1,588 @@
+import javax.swing.filechooser.FileNameExtensionFilter;
+
+import DATA.Classroom;
+import DATA.DataStore;
+import DATA.Group;
+import DATA.Teacher;
+
+public class MainFrame extends JFrame implements ActionListener {
+
+	private JPanel contentPane;
+	private DataStore dataStore;
+	private JButton btnSave;
+	private JButton btnOpen;
+	private JButton btnNew;
+	private JButton btnTeachers;
+	private AbstractButton btnClassrooms;
+	private JButton btnGroups;
+	private JButton btnFields;
+	private JButton btnWeekTable;
+	private JButton btnLancer;
+	private JButton btnFillFrame;
+	private JPanel listPanel;
+	private JPanel listPanel2;
+	private JPanel EdTPanel;
+	
+	public MainFrame(DataStore ds) {
+
+		setDefaultCloseOperation(JFrame.EXIT_ON_CLOSE);
+		setBounds(100, 100, 450, 300);
+		
+		this.dataStore = ds;
+		
+		contentPane = new ContentPane();
+		contentPane.setBorder(new EmptyBorder(5, 5, 5, 5));
+		contentPane.setLayout(new BorderLayout(0, 0));
+		setContentPane(contentPane);
+		setExtendedState(JFrame.MAXIMIZED_BOTH);
+		setVisible(true);
+		setResizable(true);
+		
+		JToolBar toolBar = new JToolBar();
+		toolBar.setFloatable(false);
+		toolBar.setMargin(new Insets(5, 5, 5, 5));
+		toolBar.setBackground(new Color(230, 230, 230));
+		getContentPane().add(toolBar, BorderLayout.NORTH);
+		
+		try {
+			btnSave = new JButton(new ImageIcon(ImageIO.read(new File("img/icons/disk.png"))));
+			btnSave.addActionListener(new ActionListener() {
+				public void actionPerformed(ActionEvent e) {
+					BufferedReader br;
+					try {
+						br = new BufferedReader(new FileReader(System.getProperty("user.dir") + "/dataStoreLocation.loc"));
+						String fileName = "";
+						String sCurrentLine = "";
+						while ((sCurrentLine = br.readLine()) != null) {
+							fileName = sCurrentLine;
+						}
+						File fichier =  new File(fileName) ;
+						 // ouverture d'un flux sur un fichier
+						ObjectOutputStream oos =  new ObjectOutputStream(new FileOutputStream(fichier)) ;
+
+						 // sérialization de l'objet
+						oos.writeObject(dataStore);
+				    	JOptionPane.showMessageDialog(null, "La sauvegarde a été réalisée avec succès.", "Sauvegarde réussie", JOptionPane.INFORMATION_MESSAGE);
+
+					} catch (IOException e1) {
+						// TODO Auto-generated catch block
+						e1.printStackTrace();
+					}
+				}
+			});
+			btnSave.setText("Enregistrer");
+			
+			btnOpen = new JButton(new ImageIcon(ImageIO.read(new File("img/icons/folder.png"))));
+			btnOpen.setText("Ouvrir");
+			btnOpen.addActionListener(new ActionListener() {
+				
+				@Override
+				public void actionPerformed(ActionEvent e) {
+int n = JOptionPane.showConfirmDialog(null, "Voulez sauvegarder la session en cours ?", "Attention", JOptionPane.YES_NO_CANCEL_OPTION);
+					
+					switch(n)	{
+					case JOptionPane.YES_OPTION :
+						BufferedReader br;
+						try {
+							br = new BufferedReader(new FileReader(System.getProperty("user.dir") + "/dataStoreLocation.loc"));
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code might contain a resource leak. Resource leaks can cause your system to slow down or crash.

**Fix**
Consider closing the following resource: *br*. Currently, there are execution paths that do not contain closure statements, for example, when an else condition at line 164 evaluates to true. Ensure *br* is closed in a try-finally block or declared in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).